modular/gater/metadata_handler_test.go

```diff
@@ -0,0 +1,2083 @@
+package gater
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"encoding/xml"
+	"fmt"
+	"log"
+	"math/big"
+	"net/http"
+	"net/http/httptest"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"cosmossdk.io/math"
+	"github.com/bnb-chain/greenfield-storage-provider/modular/metadata/types"
```

Comment:
Considering grouping github.com/bnb-chain/greenfield-storage-provider/modular/metadata/types with github.com/bnb-chain/greenfield-storage-provider/base/gfspclient for better modularity and organization.